backend/views/core/settings/view.py

```diff
@@ -27,8 +29,18 @@ def settings_page(request: HttpRequest):
                 messages.error(request, "Invalid currency")
 
         if profile_picture:
```

Comment:
I believe this should be swapped with
```py
        if profile_picture:
            try:
                if not profile_picture:
                    raise ValidationError("A profile picture is required")
                usersettings.full_clean()
                validate_image_file_extension(profile_picture)
            except ValidationError:
                messages.error(request, "Invalid profile picture file type")
            else:  # This runs only if the profile picture doesn't cause any exceptions
                usersettings.profile_picture = profile_picture
                usersettings.save()
                messages.success(request, "Successfully changed your profile picture")
```

Reply:
Now that I see it, the if statement below the try block `if not profile_picture:
                    raise ValidationError("A profile picture is required") ` will never be executed because, in the first if statement, a picture must be uploaded first.

Furthermore, I did some research and found that **validate_image_file_extension** is based on PIL and checks for a 
**valid image extension**. This includes files that can support an image like PDF, MPEG, and others without the option of adding a parameter to include only the wanted file extensions.

Reply:
>  if statement below the try block will never be executed
Yeah that's a good point, i thought of that when writing it lmao. Sorry that was a bit of a stupid snippet.

> and others without the option of adding a parameter to include only the wanted file extensions.

Okay cool, so would you suggest we use validate_image_file_extension or just use PIL ourself? As long as its safe that's fine, just don't want to miss any vulnerabilities or "make my own solution to a problem thats been fixed by someone smarter" is my kind of thoughts. I always try and re-invent the wheel and mess up.

But if you think its okay i don't mind.

Sorry about the late response. If you have discord and would prefer to speak there let me know at `ospam` if not it's okay here.

Thanks again :)

Reply:
Hey @TreyWW right now I'm doing some research whether PIL is enough for security reason. Otherwise, I think we need to use a more "official" Django way.

My discord is "radiatedfox" feel free to send me an invite :)

Reply:
Thanks for the reply, adding now :)